core/opendaq/reader/src/block_reader_impl.cpp

```diff
@@ -221,14 +286,19 @@ ErrCode BlockReaderImpl::readPackets(IReaderStatus** status, SizeT* count)
         {
             case PacketType::Data:
             {
-                info.dataPacket = packet;
+                if (info.currentDataPacketIter == info.dataPacketsQueue.end())
+                {
+                    info.dataPacketsQueue.emplace_back(packet);
+                    info.currentDataPacketIter = --info.dataPacketsQueue.end();
+                }
                 errCode = readPacketData();
 
                 if (OPENDAQ_FAILED(errCode))
                 {
+                    auto writtenSamplesCount = calculateWrittenSamplesCount(initialWrittenSamplesCount, info.writtenSampleCount);
                     if (status)
-                        *status = BlockReaderStatus(nullptr, true, (samplesToRead - info.remainingSamplesToRead) % blockSize).detach();
-                    *count = (samplesToRead - info.remainingSamplesToRead) / blockSize;
+                        *status = BlockReaderStatus(nullptr, true, writtenSamplesCount).detach();
+                    *count = writtenSamplesCount % blockSize;
```

Comment:
This should be division, not modulo if I'm not mistaken.

Reply:
Right. Fixed